.github/workflows/publish-java-cdk-command.yml

```diff
@@ -0,0 +1,249 @@
+        uses: actions/checkout@v3
+        with:
+          repository: airbytehq/airbyte
+          ref: master
+      - name: Check PAT rate limits
+        run: |
+          ./tools/bin/find_non_rate_limited_PAT \
+            ${{ secrets.GH_PAT_BUILD_RUNNER_OSS }} \
+            ${{ secrets.GH_PAT_BUILD_RUNNER_BACKUP }}
+      - name: Start AWS Runner
+        id: start-ec2-runner
+        uses: ./.github/actions/start-aws-runner
+        with:
+          aws-access-key-id: ${{ secrets.SELF_RUNNER_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.SELF_RUNNER_AWS_SECRET_ACCESS_KEY }}
+          github-token: ${{ env.PAT }}
+          label: ${{ github.run_id }}-publisher
+
+  publish-cdk:
+    name: Publish Java CDK
+    needs: start-publish-docker-image-runner-0
+    runs-on: ubuntu-latest
+    steps:
+      - name: Link comment to workflow run
+        if: github.event.inputs.comment-id
+        uses: peter-evans/create-or-update-comment@v1
+        with:
+          comment-id: ${{ github.event.inputs.comment-id }}
+          body: |
+            > :clock2: https://github.com/${{github.repository}}/actions/runs/${{github.run_id}}
+      - name: Checkout Airbyte
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ env.REPO }}
+          ref: ${{ env.GITREF }}
+      - name: Read Target Java CDK version
+        id: read-target-java-cdk-version
+        run: |
+          cdk_version=$(cat $CDK_VERSION_FILE_PATH | tr -d '\n')
+          if [[ -z "$cdk_version" ]]; then
+            echo "Failed to retrieve CDK version from $CDK_VERSION_FILE_PATH"
+            exit 1
+          fi
+          echo "CDK_VERSION=${cdk_version}" >> $GITHUB_ENV
+      - name: Check for already-published version (${{ env.CDK_VERSION }}, FORCE=${{ env.FORCE }})
+        if: ${{ !(env.FORCE == 'true') }}
+        run: |
+          GROUP_ID="io.airbyte"
+          ARTIFACT_ID="airbyte-cdk"
+          VERSION=${CDK_VERSION}
+          REPO_URL="https://airbyte.mycloudrepo.io/repositories/airbyte-public-jars"
+          GROUP_ID_URL=$(echo $GROUP_ID | tr '.' '/')
+          URL="${REPO_URL}/${GROUP_ID_URL}/${ARTIFACT_ID}/${VERSION}/${ARTIFACT_ID}-${VERSION}.pom"
+          echo "Checking if URL exists: ${URL}"
+          if curl --output /dev/null --silent --head --fail "$URL"; then
+            echo "Version ${VERSION} of ${ARTIFACT_ID} has already been published. Aborting."
+              exit 1
+          else
+              echo "Version ${VERSION} of ${ARTIFACT_ID} has not been published. Proceeding..."
+          fi
+      - name: Setup Java
+        uses: actions/setup-java@v3
+        with:
+          distribution: "zulu"
+          java-version: "17"
+      - name: Build Java CDK
+        run: ./gradlew --no-daemon :airbyte-cdk:java:airbyte-cdk:build
+      - name: Publish Java Modules to LocalMaven (Dry-Run)
+        run: ./gradlew --no-daemon :airbyte-cdk:java:airbyte-cdk:publishToMavenLocal
```

Comment:
@aaronsteers is `publishToMavenLocal` a prereq to `publish` or can we skip this step if `DRY_RUN` is false?

Reply:
We could absolutely skip if we want to. I didn't see any harm to it running in both scenarios though. If you'd like, I can toggle it off when a dry-run is not set

Reply:
Ah gotcha! Then my only minor nit pick is we should skip this step.

My thinking behind that is
1. we cant really debug these github actions without heavily modifiying these workflow files, and even then its a nightmare
2. would give us the smallest speed savings when publishing 🏎️

Reply:
Done 👍